src/solver/expressions/visitors/AstDOTStyleVisitor.cpp

```diff
@@ -42,6 +40,17 @@ static constexpr BoxStyle ComponentVariableStyle{"goldenrod", "octagon", "filled
 static constexpr BoxStyle PortFieldStyle{"olive", "component", "filled, solid"};
 } // namespace NodeStyle
 
+std::string makeLegendTitle()
+{
+    std::string to_return = "subgraph cluster_legend {\n";
```

Comment:
@flomnes will tell you to avoid tmp string and retrieve the stream as arg

Reply:
Done